src/validation.cpp

```diff
@@ -2441,7 +2369,7 @@ bool CChainState::ConnectTip(CValidationState& state, const CChainParams& chainp
     LogPrint(BCLog::BENCH, "  - Connect postprocess: %.2fms [%.2fs (%.2fms/blk)]\n", (nTime6 - nTime5) * MILLI, nTimePostConnect * MICRO, nTimePostConnect * MILLI / nBlocksTotal);
     LogPrint(BCLog::BENCH, "- Connect block: %.2fms [%.2fs (%.2fms/blk)]\n", (nTime6 - nTime1) * MILLI, nTimeTotal * MICRO, nTimeTotal * MILLI / nBlocksTotal);
 
-    connectTrace.BlockConnected(pindexNew, std::move(pthisBlock));
+    GetMainSignals().BlockConnected(std::move(pthisBlock), pindexNew);
```

Comment:
I can't comment on the exact line above (currently L2361), but can you update:
```
     // Remove conflicting transactions from the mempool.;
```
to:
```
     // Remove conflicting transactions from the mempool. This will fire the MempoolUpdatedForBlockConnect() notification.
```
or similar